tensorflow/tsl/platform/cpu_info.cc

```diff
@@ -22,6 +22,13 @@ limitations under the License.
 #if defined(PLATFORM_IS_X86)
 #include <mutex>  // NOLINT
 #endif
+#if defined(PLATFORM_IS_ARM64)
+#include <sys/auxv.h>
```

Comment:
Is this header needed? What function is it needed for? (I don't think `getauxval` needs this header.) 
It broke MacOS build. If it's not needed, I'll just remove it. If it is. I'll add a guard for Apple.

Reply:
HI @penpornk, in that header are HWCAP bits defined that we are using to check for presence of CPUID registers. I would suggest to add guard for Apple and also OpenBSD in case TF is build there too.

`#if !defined(__APPLE__) && !defined(__OpenBSD__) && defined(PLATFORM_IS_ARM64)`

Sorry about breakage.

Reply:
Should I just change everything about cpu info (CPUIDInfo class, etc) that has `#ifdef PLATFORM_IS_ARM64` to `#if defined(PLATFORM_IS_ARM64) && !defined(__APPLE__) && !defined(__OpenBSD__)` ? Or just guarding the header should be enough (i.e., other code should be generic enough for other platforms)?

Reply:
We have put the check `#if !defined(PLATFORM_WINDOWS) && !defined(__APPLE__) && !defined(__OpenBSD__)`  around reading files (see here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tsl/platform/cpu_info.cc#L380), but on the second thoughts I think it might be better to put as you have suggested around all instances of `PLATFORM_IS_ARM64` in `cpu_info.cc` because in `mkl_heuristics.h` we are using `tsl::port::CPUFamily()` and `tsl::port::CPUModelName()` and that will return 0 on platforms that do not confirm to `ifdef`.

Thanks!